### ISSUE TITLE
Push tile keyboard shortcuts have extra steps

### ISSUE THREAD

**Issue by fs2597** [linuxmint/cinnamon#12086]
### Distribution

Mint 21.3

### Package version

6.0.4

### Graphics hardware in use

NVIDIA GeForce GTX 1060

### Frequency

Always

### Bug description

The "Push tile left" and "Push tile right" keyboard shortcuts now take two tries to actually move a window from side to side. If it's snapped to the left edge, I have to "Push tile right" twice to snap it to the right edge. The first time "restores" the window's "original" size, and only the second one actually pushes it to the place I told it to go. Same thing going right to left.

### Steps to reproduce

1. Win+E to open a Cinnamon window (works as expected)
2. Win+Left to "push tile left" (works as expected)
3. Win+Right to "push tile right"

### Expected behavior

What I expect to happen: The window moves to take up the right half of the screen.

What actually happens: The window moves back to whatever random size and place it opened in, and I have to hit Win+Right a second time to send it to the right half of the screen.

### Additional information

This didn't happen in Mint 19.3, or any earlier version from the last 10 years.

I finally got around to upgrading the other day, and I noticed it right away after stepping up to 20.0, and it's still like that after getting all the way to 21.3.

I did a fresh 21.3 install in Virtualbox to make sure it wasn't something that got messed up in the upgrade process, but it's like that there, too.

**Comment by mtwebster** [comment]
This has been the behavior since Cinnamon 2.0, also just tested in 19.3 and LMDE4 (both Cinnamon 4.4.8)..  maybe you had made some custom solution before?

This is why the keybindings are called 'push' not just 'tile-right' or tile-left' - they let you 'push' the window around the edges, since there are no direct keybindings for corner positions. The untiled midway position between up/down and left/right is just a logical step for navigation.

**Comment by fs2597** [comment]
Nope, never had any custom solution before, just bound keys to in the built-in Keyboard > Shortcuts window. Not running anything custom for keyboard shortcuts or window management, just stock Cinnamon + Nemo.

I know up/down have always had extra steps (up toggles between top half and maximized, and the "untiled" state is in between up and down) but left/right never have, and I've used these probably 50 times a day for the last 5+ years. The behavior definitely and noticably changed as soon as I upgraded 19.3 to 20.

If it's not Cinnamon, what else could have changed in the upgrade from 19.3 to 20 to make this happen? I think my last "clean" install was Mint 18.

Or where in the code is this handled? I've never looked at C, Python, or GTK, but this is annoying enough I'd spend as much time as it takes to figure out how to patch it and build my own. :frowning_face:
